agent/acl_endpoint_test.go

```diff
@@ -1222,6 +1222,26 @@ func TestACL_LoginProcedure_HTTP(t *testing.T) {
 			methodMap[method.Name] = method
 		})
 
+		t.Run("Create in invalid datacenter", func(t *testing.T) {
+			methodInput := &structs.ACLAuthMethod{
+				Name:        "other",
+				Type:        "testing",
+				Description: "test",
+				Config: map[string]interface{}{
+					"SessionID": testSessionID,
+				},
+				TokenLocality: "global",
+				MaxTokenTTL:   500_000_000_000,
+			}
+
+			req, _ := http.NewRequest("PUT", "/v1/acl/auth-method?token=root&dc=invalid", jsonBody(methodInput))
+			resp := httptest.NewRecorder()
+			_, err := a.srv.ACLAuthMethodCRUD(resp, req)
```

Comment:
Not a blocker, just an FYI that we've started to change our tests to do this: #11396

Right now the url path on line 1237 is ignored (the query params are used though), and the test doesn't exercise any of the wrappers around endpoints.

Instead of calling `ACLAuthMethodCRUD` we are moving to call `a.srv.h.ServeHTTP`, which makes the test closer to a real API call.

Reply:
🙏 thanks for the heads up! I don't mind making this change across the file.